web/src/components/EpisodeViewer.vue

```diff
@@ -250,6 +256,35 @@ const checkProgress = async (force?: boolean) => {
     }
 }
 
+// This is a very naive v1.0 of the BMM streak tracking.
+// With this implementation, we have basically no way of preventing the
+// user from skipping forward to the end, and therefore getting the streak point.
+let hasSentHalfwayWatchedEvent = false
+function onVideoProgress(
+    currentTime: number,
+    duration: number,
+    player: Player
+) {
+    const videoProgressPercent = currentTime / duration
+    if (
+        (videoProgressPercent >= 0.5 && !hasSentHalfwayWatchedEvent) ||
+        videoProgressPercent === 1
```

Comment:
how often is this function called with videoProgressPercent === 1?
If it's more than once, we should make sure to send the event only once.

Reply:
Based on my testing, it is called only once. When `currentTime === duration`, the player stops playing

Reply:
ok. if you're confident in that this can be merged

Reply:
it could make sense to add a comment though